persistence/sql/sql_process_store.go

```diff
@@ -62,20 +63,92 @@ func (p sqlProcessStoreImpl) doStartProcessTx(
 	prcExeId := uuid.MustNewUUID()
 	hasNewWorkerTask := false
 
-	err := tx.InsertCurrentProcessExecution(ctx, extensions.CurrentProcessExecutionRow{
+	if req.ProcessStartConfig == nil {
+		return nil, fmt.Errorf("processStartConfig is required")
+	}
+
+	prevProcessExecution, errGetPrevExecution := tx.SelectProcessExecutionForUpdate(ctx, prcExeId)
```

Comment:
This is a new executionId, there is no execution existing yet for this id.

Reply:
looks like you wanted to get the prevExecution, it should be after selectCurrentExecution

Reply:
if there is a running process with the same id, and id reuse policy is terminate_if_running, we still need the previous execution, right?

Reply:
Yes, to get the previous execution, we need to get the processExeucitonID from selectCurrentExecution. Doing it before selectCurrentExecution will have no value? 

Note that the processExeucitonID is uniquely created for every request.